src/components/NamespacedPage/NamespacedPage.tsx

```diff
@@ -9,23 +8,21 @@ import SignupView from '../Signup/SignupView';
 
 import './NamespacedPage.scss';
 
-type NamespaceContextData = {
-  namespace: string;
-};
+type NamespaceContextData = [string, boolean];
+
+const NamespaceContext = React.createContext<NamespaceContextData>(['', false]);
 
-export const NamespaceContext = React.createContext<NamespaceContextData>({
-  namespace: '',
-});
+export const NamespaceProvider = NamespaceContext.Provider;
 
-export const useNamespace = () => React.useContext(NamespaceContext);
+export const useNamespace = () => React.useContext(NamespaceContext)[0];
```

Comment:
If you have the information of loaded or not maybe you should use it here in the hook somehow. What happens when someone tries to get the namespace from this hook and its not loaded yet?

Reply:
Our NamespacedPage Component handles this case. if namespace is not loaded we show a spinner. That's why it returns namespace string only.